### PR TITLE
Feature/clickable mixin

### PR DESCRIPTION
**Pull request by jthrilly**
This is a fun little idea I had as part of thinking about the lack of tactility of our flat designs. 

Basically it adds a mixin (`clickable`) that implements a box-shadow on an element, a hover effect that grows the box-shadow, and an active effect that collapses it. This overall gives the sense of depth and tactility. 

I also added a depth parameter to the mixin, so that for larger elements like cards and buttons, you can get an appropriately large shadow.

For now, this is only added to the button element.

Better implementations and feedback welcome. I did attempt to implement this: https://tobiasahlin.com/blog/how-to-animate-box-shadow/
 
...but couldn't get it to work. 

Screen Recording 2020-04-24 at 12 57 25 PM

### CODE REVIEW

**Comment by wwqrd** [comment]
Just for reference I implemented it using the technique you mentioned (on https://tobiasahlin.com/blog/how-to-animate-box-shadow/) -- it has the following drawbacks:

- Parent element must have a position context (e.g. absolute or relative)
- Mixin would need to know about border radius (have hard coded it in the reference implementation)

https://github.com/codaco/Network-Canvas-UI/tree/feature/clickable-mixin-performant

File: README.md [hunk 1]
```diff
@@ -17,7 +17,17 @@ npm install --save @codaco/ui
...
-## Usage
+## Development
+
+In this repo:
+
+`npm link` then in consuming app ``
+
+After each change:
+
+`npm install; npm run build; npm install --only=production`
```
Comment (wwqrd) [inline README.md:28]: Should just be able to do a build in the local UI directory (which will now be symlinked), unless there are packages that have changed?

File: README.md [hunk 2]
```diff
@@ -17,7 +17,17 @@ npm install --save @codaco/ui
 }
 ```
 
-## Usage
+## Development
+
+In this repo:
+
+`npm link` then in consuming app ``
```
Comment (wwqrd) [inline README.md:24]: > then in consuming app `npm link @codaco/ui`

File: README.md (hunk 1 above)
Comment (jthrilly) [reply README.md:28]: Ah. This was to do with the react hooks errors I was getting when I tried this. The above command was one you suggested, and it has been the only thing that works for me!

Comment (jthrilly) [reply README.md:28]: To be clear: this is only when I am developing on UI and importing via npm link into NC - it could be different for Architect maybe?

**Pull request merged**